packages/flutter/lib/src/gestures/scale.dart

```diff
@@ -410,12 +412,26 @@ class ScaleGestureRecognizer extends OneSequenceGestureRecognizer {
   void _update() {
     final int count = _pointerLocations.keys.length;
 
+    final Offset? previousFocalPoint = _state != _ScaleState.started ? null : _currentFocalPoint;
+
     // Compute the focal point
     Offset focalPoint = Offset.zero;
     for (final int pointer in _pointerLocations.keys)
       focalPoint += _pointerLocations[pointer]!;
     _currentFocalPoint = count > 0 ? focalPoint / count.toDouble() : Offset.zero;
 
+    final Offset localFocalPoint = PointerEvent.transformPosition(
+      _lastTransform,
+      _currentFocalPoint,
+    );
+    final Offset localPreviousFocalPoint = _state != _ScaleState.started
+        ? localFocalPoint
+        : PointerEvent.transformPosition(
+            _lastTransform,
```

Comment:
if the receiving render object moved then `_lastTransform` is also changed right (I'm assuming `_lastTransform` is the transform from the global coordinate space to the local coordinate space). Should we use a local variable to track `localFocalPoint`?

Reply:
Done, good idea.

Reply:
You don't have to compute `localPreviousFocalPoint` I think? It's just `_localFocalPoint` before you updated it in line 425.

Reply:
Ah, I was working around the fact that _localFocalPoint is `late` and won't be initialized on the first run here, so I can't grab the value out of it here.

I've updated it to an if/else that avoids the recalculation, let me know if you like the old way better.